dashboard/app/controllers/api/v1/pd/workshop_survey_foorm_submissions_controller.rb

```diff
@@ -9,12 +14,53 @@ def create
       day: params[:day]
     )
     begin
-      survey_submission.save_with_foorm_submission(answers, params[:form_name], params[:form_version])
+      survey_submission.save_with_foorm_submission(answers.to_json, params[:form_name], params[:form_version])
     rescue ActiveRecord::ActiveRecordError => e
       render json: {error: e.message}, status: :bad_request
       return
     end
 
-    render json: {submission_id: survey_submission.foorm_submission_id, survey_submission_id: survey_submission.id}, status: :created
+    facilitator_save_result = save_facilitator_questions(facilitator_answers, params)
+
+    if facilitator_save_result[:save_success]
+      render json: {submission_id: survey_submission.foorm_submission_id, survey_submission_id: survey_submission.id}, status: :created
+    else
+      render json: {error: e.message}, status: :bad_request
+    end
+  end
+
+  # save each set of facilitator questions (split per facilitator)
+  # as a separate survey submission
+  def save_facilitator_questions(facilitator_answers, params)
+    save_success = true
+    error = nil
+    submission_ids = []
+    survey_submission_ids = []
+    if facilitator_answers
+      facilitator_answers.each do |_, data|
+        next unless data["facilitatorId"]
+        survey_submission = ::Pd::WorkshopSurveyFoormSubmission.new(
+          user_id: params[:user_id],
+          pd_session_id: params[:pd_session_id],
+          pd_workshop_id: params[:pd_workshop_id],
+          day: params[:day],
+          facilitator_id: data["facilitatorId"]
+        )
+        begin
+          survey_submission.save_with_foorm_submission(data.to_json, params[:form_name], params[:form_version])
+          submission_ids << survey_submission.foorm_submission_id
+          survey_submission_ids << survey_submission.id
```

Comment:
Does this effectively work as a transaction

Reply:
it does because save_with_foorm_submission creates a transaction which saves the survey and the submission